src/Tickets/Commerce/Reports/Orders.php

```diff
@@ -0,0 +1,381 @@
+		if ( empty( $ticket_id ) ) {
+			return '';
+		}
+		$post = get_post( $event_id );
+
+		$query = [
+			'post_type'   => $post->post_type,
+			'page'        => static::$page_slug,
+			'product_ids' => $ticket_id,
+			'post_id'     => $event_id,
+		];
+
+		$report_url = add_query_arg( $query, admin_url( 'admin.php' ) );
+
+		return '<span><a href="' . esc_url( $report_url ) . '">' . esc_html__( 'Report', 'event-tickets' ) . '</a></span>';
+	}
+
+	/**
+	 * Returns the link to the "Orders" report for this post.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return string The absolute URL.
+	 */
+	public static function get_tickets_report_link( $post ) {
+		$url = add_query_arg( [
+			'post_type' => $post->post_type,
+			'page'      => static::$page_slug,
+			'post_id'   => $post->ID,
+		], admin_url( 'edit.php' ) );
+
+		return $url;
+	}
+
+	/**
+	 * Hooks the actions and filter required by the class.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		add_filter( 'post_row_actions', [ $this, 'add_orders_row_action' ], 10, 2 );
+		add_action( 'tribe_tickets_attendees_page_inside', [ $this, 'render_tabbed_view' ] );
+		add_action( 'admin_menu', [ $this, 'register_orders_page' ] );
+
+		// register the tabbed view
+		$paypal_tabbed_view = new \Tribe__Tickets__Commerce__PayPal__Orders__Tabbed_View();
+		$paypal_tabbed_view->register();
+	}
+
+	/**
+	 * Adds order related actions to the available row actions for the post.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $actions
+	 * @param       $post
+	 *
+	 * @return array
+	 */
+	public function add_orders_row_action( array $actions, $post ) {
+		$post_id = \Tribe__Main::post_id_helper( $post );
+		$post    = get_post( $post_id );
+
+		// only if tickets are active on this post type
+		if ( ! in_array( $post->post_type, Plugin::instance()->post_types(), true ) ) {
+			return $actions;
+		}
+		$commerce = tribe( Module::class );
+
+		if ( ! $commerce->post_has_tickets( $post ) ) {
+			return $actions;
+		}
+
+		$url         = $commerce->get_event_reports_link( $post->ID, true );
+		$post_labels = get_post_type_labels( get_post_type_object( $post->post_type ) );
+		$post_type   = strtolower( $post_labels->singular_name );
+
+		$actions['tickets_orders'] = sprintf(
+			'<a title="%s" href="%s">%s</a>',
+			sprintf( esc_html__( 'See PayPal purchases for this %s', 'event-tickets' ), $post_type ),
+			esc_url( $url ),
+			esc_html__( 'PayPal Orders', 'event-tickets' )
+		);
+
+		return $actions;
+	}
+
+	/**
+	 * Renders the tabbed view header before the report.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Tickets__Attendees $attendees
+	 */
+	public function render_tabbed_view( \Tribe__Tickets__Attendees $attendees ) {
+		$post = $attendees->get_post();
+
+		$commerce = tribe( Module::class );
+
+		$has_tickets = $commerce->get_attendees_count( $post->ID );
+		if ( ! $has_tickets ) {
+			return;
+		}
+
+		$tabbed_view = new \Tribe__Tickets__Commerce__PayPal__Orders__Tabbed_View();
```

Comment:
Shouldn't this orders report be agnostic and we should be hooking these from each gateway?